### PR TITLE
PHEE-249 Parameterize collection payment scheme

### PR DESCRIPTION
**Pull request by victorjbassey**
## Description

In addition to Mpesa, other similar flows (ex. Airtel Money, MTN MoMo, etc) can be supported by parameterizing the payment scheme.

This PR adds support for an optional `X-Payment-Scheme` header whose value will determine which bpmn flow will be started when a collection request is received.

Jira ticket: https://mifosforge.jira.com/browse/PHEE-249

@avikganguly01 @fynmanoj @ankita10r please review

### CODE REVIEW

**Comment by truthfool** [comment]
Hi @victorjbassey Can you create a JIRA Ticket for the pull request. It will not be reviewed/merged unless a Jira Ticket is present. If you are not able to create a JIRA ticket pls reach out to @fynmanoj or @avikganguly01 .

**Comment by victorjbassey** [comment]
> Hi @victorjbassey Can you create a JIRA Ticket for the pull request. It will not be reviewed/merged unless a Jira Ticket is present. If you are not able to create a JIRA ticket pls reach out to @fynmanoj or @avikganguly01 .

Hi @truthfool 
Can you share the link to where I can create the ticket?

**Comment by truthfool** [comment]
https://mifosforge.jira.com/jira/software/c/projects/PHEE/issues

File: src/main/java/org/mifos/connector/channel/camel/config/CamelProperties.java [hunk 1]
```diff
@@ -10,5 +10,6 @@ private CamelProperties() {}
     public static final String CLIENTCORRELATIONID = "X-CorrelationID";
     public static final String BATCH_ID_HEADER = "X-BatchID";
     public static final String PLATFORM_TENANT_ID = "Platform-TenantId";
+    public static final String PAYMENT_SCHEME_HEADER = "X-PaymentScheme";
```
Comment (avikganguly01) [inline src/main/java/org/mifos/connector/channel/camel/config/CamelProperties.java:13]: API Guidelines indicate that new headers have to follow hyphen separated naming convention like X-Payment-Scheme.

**Review by avikganguly01 — CHANGES_REQUESTED** [review]

**Review by avikganguly01 — CHANGES_REQUESTED** [review]
@Victor Bassey :
The pipelines have ran and they are successful. Please follow coding conventions by following this guide - https://github.com/apache/fineract#checkstyle-and-spotless 
Use the gradle license plugin to add license header to any newly added files in future PRs. 

Please write an [integration test](https://github.com/openMF/ph-ee-integration-test) to prevent others from breaking your code. Scope of what has to be deployed in order for your integration test to pass has to be carefully designed. Ex:- If test case is dependent on a specific AMS which is not deployed in CircleCI scripts, the test case will fail. 

Perhaps a simple scope could be an unit test instead of an integration test here to see if PS is resolved which is a prerequisite for orchestration configuration to be resolved.

**Comment by victor** [comment]
@avikganguly01 please watch your mentions. thanks

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**